src/main/java/seedu/address/logic/commands/AddEventCommand.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INTERVAL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REPEATABLE_DATE;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.project.EventList;
+import seedu.address.model.project.Project;
+import seedu.address.model.task.repeatable.Event;
+
+public class AddEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "addEto";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the event list of the project identified "
+            + "by the index number used in the displayed project list. "
+            + "Existing list will be appended with the input event.\n"
+            + "Parameters: PROJECT_INDEX (must be a positive integer) "
+            + "[" + PREFIX_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_INTERVAL + "INTERVAL] "
+            + "[" + PREFIX_REPEATABLE_DATE + "REPEATABLE_DATE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_DESCRIPTION + "Project meeting"
+            + PREFIX_INTERVAL + "DAILY"
+            + PREFIX_REPEATABLE_DATE + "24-04-2021";
+
+    public static final String MESSAGE_EVENT_ADDED_SUCCESS = "New event added: %1$s";
+
+    private final Index index;
+    private final Event eventToAdd;
+
+
+    /**
+     * Initializes a addEventCommand.
+     *
+     * @param index index of project to add event.
+     * @param event event to add.
+     */
+    public AddEventCommand(Index index, Event event) {
+        requireNonNull(index);
+        requireNonNull(event);
+
+        this.index = index;
+        this.eventToAdd = event;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Project> lastShownList = model.getFilteredProjectList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
```

Comment:
great! for anyone else checking getZeroBased() method checks for negative indexes alr :)